app/models/claim.rb

```diff
@@ -554,6 +554,54 @@ def recent_tps_school
     TeachersPensionsService.recent_tps_school(self)
   end
 
+  # dup - because we don't want to pollute the claim.errors by calling this method
+  # Used to not show the personal-details page if everything is all valid
+  def has_all_valid_personal_details?
+    dup.valid?(:"personal-details") && all_personal_details_same_as_tid?
+  end
+
+  # This is used to ensure we still show the forms if the personal-details are valid
+  # but are valid because they were susequently provided/changed from what was in TID
+  def all_personal_details_same_as_tid?
+    name_same_as_tid? && dob_same_as_tid? && nino_same_as_tid?
+  end
+
+  def name_same_as_tid?
+    teacher_id_user_info["given_name"] == first_name && teacher_id_user_info["family_name"] == surname
+  end
+
+  def dob_same_as_tid?
+    teacher_id_user_info["birthdate"] == date_of_birth.to_s
+  end
+
+  def nino_same_as_tid?
+    teacher_id_user_info["ni_number"] == national_insurance_number
+  end
+
+  # dup - because we don't want to pollute the claim.errors by calling this method
+  # Check errors hash for key because we don't care about the non-context validation errors
+  def has_valid_name?
+    claim_dup = dup
+    claim_dup.valid?(:"personal-details-name")
+    !(claim_dup.errors.include?(:first_name) || claim_dup.errors.include?(:surname))
+  end
+
+  # dup - because we don't want to pollute the claim.errors by calling this method
+  # Check errors hash for key because we don't care about the non-context validation errors
+  def has_valid_date_of_birth?
+    claim_dup = dup
+    claim_dup.valid?(:"personal-details-dob")
+    !claim_dup.errors.include?(:date_of_birth)
+  end
+
+  # dup - because we don't want to pollute the claim.errors by calling this method
+  # Check errors hash for key because we don't care about the non-context validation errors
+  def has_valid_nino?
+    claim_dup = dup
+    claim_dup.valid?(:"personal-details-nino")
+    !claim_dup.errors.include?(:national_insurance_number)
+  end
```

Comment:
nit: maybe we could extract this out into another model/service

Reply:
Good point, didn't think to split it out as this thing is already massive!